skbio/sequence/_sequence.py

```diff
@@ -1663,10 +1664,22 @@ def iter_kmers(self, k, overlap=True):
         if k < 1:
             raise ValueError("k must be greater than 0.")
 
-        step = 1 if overlap else k
+        if overlap:
+            step = 1
+            count = len(self) - k + 1
+        else:
+            step = k
+            count = len(self) // k
 
-        for i in range(0, len(self) - k + 1, step):
-            yield self[i:i+k]
+        if self.has_positional_metadata():
+            for i in range(0, len(self) - k + 1, step):
+                yield self[i:i+k]
+        # Optimized path when no positional metadata
+        else:
+            kmers = as_strided(self._bytes, shape=(k, count),
+                               strides=(1, step)).T
+            for s in kmers:
+                yield self._constructor(sequence=s)
```

Comment:
This will drop everything and only include the sequence. This is inconsistent with how the unoptimized path behaves via `__getitem__`. I think changing this to use `self._to` should fix it.